core/src/tx.rs

```diff
@@ -169,32 +214,40 @@ impl TransactionValidator {
                     .map_err(|reason| WasmExecutionFail {
                         reason: reason.to_string(),
                     })
-                    .map_err(TransactionRejectionReason::WasmExecution)?;
+                    .map_err(TransactionRejectionReason::WasmExecution)
             }
         }
-
-        Ok(())
     }
 
-    fn validate_signatures(
-        &self,
-        tx: &AcceptedTransaction,
-        is_genesis: bool,
+    fn validate_with_runtime_validators(
+        tx: AcceptedTransaction,
+        wsv: &WorldStateView,
     ) -> Result<(), TransactionRejectionReason> {
-        if !is_genesis && tx.payload().account_id == AccountId::genesis() {
-            return Err(TransactionRejectionReason::UnexpectedGenesisAccountSignature);
-        }
+        let AcceptedTransaction {
+            payload,
+            signatures,
+        } = tx;
+        let signatures = signatures.into_iter().collect();
 
-        let option_reason = match tx.check_signature_condition(&self.wsv) {
-            Ok(MustUse(true)) => None,
-            Ok(MustUse(false)) => Some("Signature condition not satisfied.".to_owned()),
-            Err(reason) => Some(reason.to_string()),
-        }
-        .map(|reason| UnsatisfiedSignatureConditionFail { reason })
-        .map(TransactionRejectionReason::UnsatisfiedSignatureCondition);
+        let pure_tx = Transaction {
+            payload,
+            signatures,
+        };
```

Comment:
I just realized that this seems to go against out restrictions about not using `Transaction`.
We have discussion about this in #2704 recently.
And imo seems reasonable that validator get `Accepted` transaction, that pass basic checks. 
Problem is that `AcceptedTransaction` currently defined in `core` not in `data_model`.
What do you think?

Reply:
I agree with you. But I don't know how we can fix this problem.

Maybe we could move `AcceptedTransaction` inside `data_model` and use it. Then we will have its methods inside `core` implemented with a triat.

Another question: *Why is Transaction considered Accepeted, when it wasn't checked by Validators yet?*. Maybe we should introduce a new state? Or let Validators receive Transaction and use it as `AcceptedTransaction` only after passing all validators?

Reply:
We need this PR merged ASAP. So I've created #2744 for this